src/discord/discord.py

```diff
@@ -46,6 +46,13 @@ async def _discord_request(
             'User-Agent' : 'DiscordBot (https://github.com/CSSS/csss-site-backend, 1.0)'
         }
     )
+    rate_limit_reset = float(result.headers["x-ratelimit-reset-after"])
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+
+    if rate_limit_remaining <= 2:
+        from time import sleep
```

Comment:
please import this function at the top of the module. This avoids the function being reimported every time the rate limit is reached. It will only be imported once ever if at top of module!

Reply:
Resolved in ea025d3